src/lib.rs

```diff
@@ -94,16 +94,6 @@
 //! # Ok(())
 //! # }
 //! ```
-#![deny(
-    missing_docs,
-    broken_intra_doc_links,
-    clippy::correctness,
-    clippy::complexity,
-    clippy::perf,
-    unsafe_code,
-    unreachable_pub
-)]
-#![cfg_attr(not(test), deny(unused))]
```

Comment:
NB: this is a tradeoff. I, for one, have found working on this codebase with these lints enabled at all times to be a major PITA and almost always end up disabling them and only fixing the lints before submitting a PR...